server/lobbyconnection.py

```diff
@@ -530,14 +539,75 @@ async def check_policy_conformity(self, player_id, uid_hash, session, ignore_res
 
         return response.get('result', '') == 'honest'
 
+    async def command_auth(self, message):
+        pub_key = config._API_JWT_PUBLIC_KEY_VALUE
+        token = message["token"]
+        unique_id = message["unique_id"]
+
+        try:
+            token = jwt.decode(token, pub_key, algorithms="RS256")
+            player_id = token["user_id"]
+        except (InvalidTokenError, KeyError):
+            raise AuthenticationError("Token signature was invalid")
+
+        new_irc_password = hexlify(os.urandom(16)).decode()
+        await self.send({
+            "command": "irc_password",
+            "password": new_irc_password
+        })
+
+        async with self._db.acquire() as conn:
+            result = await conn.execute(
+                select([t_login.c.login, t_login.c.steamid])
+                .where(t_login.c.id == player_id)
+            )
+            row = await result.fetchone()
+
+            username = row[t_login.c.login]
+            steamid = row[t_login.c.steamid]
+
+        await self.on_player_login(
+            player_id, username, new_irc_password, steamid, unique_id
+        )
+
     async def command_hello(self, message):
         login = message['login'].strip()
         password = message['password']
+        unique_id = message['unique_id']
 
         async with self._db.acquire() as conn:
-            player_id, login, steamid = await self.check_user_login(conn, login, password)
-            metrics.user_logins.labels("success").inc()
+            player_id, username, steamid = await self.check_user_login(
+                conn, login, password
+            )
+
+        await self.on_player_login(
+            player_id, username, password, steamid, unique_id
+        )
+
+    async def on_player_login(
+        self,
+        player_id: int,
+        username: str,
+        password: str,
+        steamid: int,
+        unique_id: str
+    ):
+        conforms_policy = await self.check_policy_conformity(
+            player_id, unique_id, self.session,
+            ignore_result=(
+                steamid is not None or
+                self.player_service.is_uniqueid_exempt(player_id)
+            )
+        )
+        if not conforms_policy:
+            return
```

Comment:
I see that the logic is staying and I'm not completely sure what the policy check is actually doing (making sure a single steam account has only one associated faf account, except for very old accounts?),
but does failing the policy check count as failed login attempt and if so, does it increase the `metrics.user_logins.labels("failure")` counter?

Reply:
Yea metrics are a little off, although I think this branch is not taken very often since most of the time we ignore policy responses anyways.